### PR TITLE
added log messages for RAM usage

### PR DESCRIPTION
**Pull request by jstucke**
- added log messages for RAM usage
- refactored `fact_base` to avoid code duplication

### CODE REVIEW

File: src/fact_base.py [hunk 1]
```diff
@@ -86,3 +96,13 @@ def do_self_test():
                 'Please run `alembic upgrade head` from `src` to update the schema.'
             )
             raise DbInterfaceError('Schema mismatch')
+
+    @staticmethod
```
Comment (maringuu) [inline src/fact_base.py:100]: I personally don't think private static methods make much sense.
Why not make it a function in in the module?

File: src/start_fact_backend.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #! /usr/bin/env python3
-'''
+"""
```
Comment (maringuu) [inline src/start_fact_backend.py:2]: I'd be really happy if we could refrain from doing unrelated ruff fixes in commits.
All those fixes will eventually (when #987 is merged) land in a single commit in #1045

**Review by maringuu — CHANGES_REQUESTED** [review]

**Review by maringuu — CHANGES_REQUESTED** [review]
See above.

File: src/fact_base.py [hunk 3]
```diff
@@ -61,17 +61,27 @@ def shutdown_listener(self, signum, _):
...
+        self.start()
         logging.info(f'Successfully started {self.PROGRAM_NAME}')
+        counter = 0
         while self.run:
-            self.work_load_stat.update()
+            self._update_component_workload()
             sleep(5)
             if self.args.testing:
                 break
+            if not (counter := counter + 1) % 12:  # only check every minute
```
Comment (maringuu) [inline src/fact_base.py:83]: I personally don't like the walrus operator. While I see its usecase in large list comprehensions to improve performance it doesn't improve performance here.
I find this very readable.

**Review by maringuu — CHANGES_REQUESTED** [review]

File: src/fact_base.py (hunk 1 above)
Comment (jstucke) [reply src/fact_base.py:100]: converted the class method to a function

File: src/start_fact_backend.py (hunk 2 above)
Comment (jstucke) [reply src/start_fact_backend.py:2]: merged master after the merge of #1045

File: src/fact_base.py (hunk 3 above)
Comment (jstucke) [reply src/fact_base.py:83]: > I find this very readable.

What is the problem then? :sweat_smile: 
Fun aside, I agree that it's harder to read than without the walrus operator -> changed

**Comment by jstucke** [comment]
> See above.

What line should be referenced here? `_is_main_process()`?
Changed that as well.

**Comment by maringuu** [comment]
> What line should be referenced here? _is_main_process()?

This was just GitHub messing around and not letting me submit a review that requests changes without also leaving a comment.

**Review by maringuu — APPROVED** [review]
